cloud-controller-manager/do/loadbalancers.go

```diff
@@ -846,8 +846,12 @@ func buildForwardingRules(service *v1.Service) ([]godo.ForwardingRule, error) {
 func buildForwardingRule(service *v1.Service, port *v1.ServicePort, protocol, certificateID string, tlsPassThrough bool) (*godo.ForwardingRule, error) {
 	var forwardingRule godo.ForwardingRule
 
-	if port.Protocol != portProtocolTCP {
-		return nil, fmt.Errorf("only TCP protocol is supported, got: %q", port.Protocol)
+	switch port.Protocol {
+	case portProtocolTCP:
+	case portProtocolUDP:
+		protocol = protocolUDP
```

Comment:
I wonder if we should do this assignment in the calling `buildForwardingRules()` function where we also assign the other protocol values. This would keep the logic that sets the protocol in one place.

It's a small bit but could make the code slightly easier to understand.

Reply:
Moved it!